pkgs/development/libraries/irrlichtmt/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, zlib
+, libpng
+, libjpeg
+, libGL
+, libX11
+, libXxf86vm
+, withTouchSupport ? false
+, libXi
+, libXext
+, Cocoa
+, Kernel
+}:
+stdenv.mkDerivation rec {
+  pname = "irrlichtmt";
+  version = "1.9.0mt4";
+
+  src = fetchFromGitHub {
+    owner = "minetest";
+    repo = "irrlicht";
+    rev = version;
+    sha256 = "sha256-YlXn9LrfGkjdb8+zQGDgrInolUYj9nVSF2AXWFpEEkw=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  # https://github.com/minetest/minetest/pull/10729
+  postPatch = lib.optionalString withTouchSupport ''
+    substituteInPlace include/IrrCompileConfig.h \
+      --replace '//#define _IRR_LINUX_X11_XINPUT2_' '#define _IRR_LINUX_X11_XINPUT2_'
```

Comment:
Could also be a compile flag (-D_IRR...), but this is fine.

Reply:
Upstream doesn't have a Cmake option for this, and [the build script](https://gist.github.com/TheBrokenRail/d1ef86d115fa8012cff13af350c5fdfd) from [the PR](https://github.com/minetest/minetest/pull/10729) that enabled touch in Minetest pretty much does this.

Reply:
Yeah, it was just a suggestion. This is fine too.